src/PolySharp.SourceGenerators/PolyfillsGenerator.TypeForwards.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using PolySharp.SourceGenerators.Extensions;
+using PolySharp.SourceGenerators.Helpers;
+
+namespace PolySharp.SourceGenerators;
+
+/// <inheritdoc/>
+partial class PolyfillsGenerator
+{
+    /// <summary>
+    /// The collection of fully qualified type names for types that could require a <c>modreq</c>.
+    /// </summary>
+    private static readonly ImmutableArray<string> ModreqCandidateFullyQualifiedTypeNames = ImmutableArray.Create(
+        "System.Index",
+        "System.Range",
```

Comment:
These two are not actually caused by modreqs, but instead because someone could realistically use the type in a public API when they are exposed as public, which could cause a problem (meaning they only need to be forwarded when public probably, and the list should probably be renamed or split).

Reply:
Fixed in https://github.com/Sergio0694/PolySharp/pull/42/commits/1436ecdd2b9b06f3bc6fe964ca6285cfe26b3bbc, each type is now properly only forwarded if (1) it's possibly generated, and (2) for `Index` and `Range`, only if they're public.